app/src/main/java/com/muzima/view/SHRObservationsDataActivity.java

```diff
@@ -0,0 +1,249 @@
+    private Button yesOptionShrSearchButton;
+    private Button noOptionShrSearchButton;
+
+    @Override
+    protected void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+        setContentView(R.layout.activity_shr__observations__data_);
+
+        // Show the Up button in the action bar.
+        setupActionBar();
+        initPager();
+        initPagerIndicator();
+    }
+
+    private void initPagerIndicator() {
+        pagerTabsLayout = (PagerSlidingTabStrip) findViewById(R.id.pager_indicator);
+        pagerTabsLayout.setTextColor(Color.WHITE);
+        pagerTabsLayout.setTextSize((int) getResources().getDimension(R.dimen.pager_indicator_text_size));
+        pagerTabsLayout.setSelectedTextColor(getResources().getColor(R.color.tab_indicator));
+        pagerTabsLayout.setTypeface(Fonts.roboto_medium(this), -1);
+        pagerTabsLayout.setViewPager(viewPager);
+        viewPager.setCurrentItem(0);
+        pagerTabsLayout.markCurrentSelected(0);
+    }
+
+    private void initPager() {
+        viewPager = (ViewPager) findViewById(R.id.pager);
+        observationsPagerAdapter = new ObservationsPagerAdapter(getApplicationContext(), getSupportFragmentManager(), IS_SHR_DATA);
+        observationsPagerAdapter.initPagerViews();
+        viewPager.setAdapter(observationsPagerAdapter);
+    }
+
+    /**
+     * Set up the {@link android.support.v7.app.ActionBar}.
+     */
+    private void setupActionBar() {
+        patient = (Patient) getIntent().getSerializableExtra(PatientSummaryActivity.PATIENT);
+        getSupportActionBar().setTitle(patient.getSummary());
+    }
+
+    @Override
+    public boolean onCreateOptionsMenu(Menu menu) {
+        getMenuInflater().inflate(R.menu.shr_observations_list, menu);
+        SearchView searchView = (SearchView) menu.findItem(R.id.search)
+                .getActionView();
+        searchView.setQueryHint(getString(R.string.info_observation_search));
+        searchView.setOnQueryTextListener(observationsPagerAdapter);
+        return true;
+    }
+
+    @Override
+    public boolean onOptionsItemSelected(MenuItem item) {
+        switch (item.getItemId()) {
+            case R.id.obs_shr_card_write:
+                prepareWriteToCardOptionDialog(getApplicationContext());
+                writeShrDataOptionDialog.show();
+                break;
+            default:
+                break;
+        }
+        return super.onOptionsItemSelected(item);
+    }
+
+    @Override
+    public void onBackPressed() {
+        super.onBackPressed();
+        observationsPagerAdapter.cancelBackgroundQueryTasks();
+    }
+
+    public void invokeShrApplication() {
+        SmartCardController smartCardController = ((MuzimaApplication) getApplicationContext()).getSmartCardController();
+        SmartCardRecord smartCardRecord = null;
+        try {
+            smartCardRecord = smartCardController.getSmartCardRecordByPersonUuid(patient.getUuid());
```

Comment:
should the function be getSmartCardRecordByPatientUuid instead of getSmartCardRecordByPersonUuid